src/OpenTelemetry.AutoInstrumentation.Loader/Startup.NetCore.cs

```diff
@@ -61,7 +61,7 @@ private static Assembly AssemblyResolve_ManagedProfilerDependencies(object sende
         // 2) The AssemblyVersion is lower than the version used by OpenTelemetry.AutoInstrumentation, the assembly will fail to load
         //    and invoke this resolve event. It must be loaded in a separate AssemblyLoadContext since the application will only
         //    load the originally referenced version
-        if (assemblyName.Name.StartsWith("OpenTelemetry.AutoInstrumentation", StringComparison.OrdinalIgnoreCase) && File.Exists(path))
+        if (assemblyName.Name != null && assemblyName.Name.StartsWith("OpenTelemetry.AutoInstrumentation", StringComparison.OrdinalIgnoreCase) && File.Exists(path))
```

Comment:
why not the `?.` operator?

Reply:
StartsWith returns `bool`, ?.StartWith returns lets say `bool?` which is not valid for the condition.